srcs/runners_manager/runner/Manager.py

```diff
@@ -135,7 +135,7 @@ def manage_runners(self):
                 manager.delete_runner(runner)
 
             # Create if it's still not enough
-            while self.need_new_runner(manager):
+            if self.need_new_runner(manager):
```

Comment:
What's the return type/value of `need_new_runner` and why changing to `if` has an impact?

Also, the `while` must have been set for a reason, to keep creating runners while they are needed. Is there a behavior impact on the code?

Reply:
the `need_new_runner` returns a bool. the impact is that the function would break out of the loop to avoid the  `create_runner`  which is called in the loop continously. the refresh endpoint would called the `runner_manager` method every two minutes. There is no impact in the sense that if the manager has reach `max` no runner would be created. It is the same in a `while` or `if` except with if we reduce the number of log entries. I was not able to reproduce the issue I commented in the ticket. Maybe I am missing something in orther to reporduce it.

Reply:
I see, when trying to reproduce the issue, were you setup on aws or google cloud?

As it seems that the gcloud runner never faced this issue (as far as I know, or we just never noticed it).

Reply:
I tried it on the gcloud runner. It is not an issue per say. Maybe due to the limited pool space in the AWS runner that is why the  `max` limit was reached allowing @gaspardmoindrot  to find the weird behavior. With the gcloud runner manager their is a many runners available at any moment so the `limit` is hardly reached. Maybe also we had the log entries but for a short period of time.

Reply:
I find it strange that you couldn't reproduce the error (whether with GCloud or AWS). How did you test it? Did you set up a pool with max=3 and bombard it with workflows, for example?

Try checking if Redis is properly initialized. It would be really helpful to find the error, as there might be something more than just a need for runners without available slots.

I'm not necessarily a fan of replacing the `while` with an `if`. My thought is that if someone needs to launch, let's say, 15 jobs in parallel, they would have to wait for 15 multiplied by 2 minutes, which is 30 minutes. I believe we can find an alternative approach that doesn't impact developers as much. Let me know what you think, @Abubakarr99.